src/main/resources/avro/common.avdl

```diff
@@ -387,19 +302,59 @@ record Position {
 }
 
 /**
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. (The bases for a sequence are
-available through the `getSequenceBases()` API call.)
+A `Join` is simply a pair of Position objects, which we call the sides. The are
+logically unordered (i.e. swapping makes no difference), but we require a rank
+on the Sequences, and so implicitly on the positions, so to avoid ambiguity we
+require that the position for side1 is less than that for side2.
+*/
+record Join {
```

Comment:
Do we need an explicit `Join` type? If it's just an ordered pair of positions, then wouldn't an `array<Position>` would be more natural (and give a more elegant programming model)? We don't actually seem to use the `Join` type anywhere except for the `SearchJoinsRequest` which seems odd.